### PR TITLE
Sample IR files generated as a result of testing.

### PR DESCRIPTION
**Pull request by amadhusu**
**Description of your changes:**
These sample IR files were generated as part of testing sample pipelines against DSPv2 at https://issues.redhat.com/browse/RHOAIENG-2880
This PR can be closed if these IR files aren't required and just unnecessary.

**Testing:**
There is nothing to test as this PR is just the IR files generated from the sample python files which are already part of this code base.

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by dsp-developers** [comment]
A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-18`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-18`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-18`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-18`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-18`
  **MLMD Server**: `quay.io/opendatahub/ds-pipelines-metadata-grpc:pr-18`
  **MLMD Envoy Proxy**: `quay.io/opendatahub/ds-pipelines-metadata-envoy:pr-18`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-18`

**Comment by dsp-developers** [comment]
An OCP cluster where you are logged in as cluster admin is required.

The Data Science Pipelines team recommends testing this using the Data Science Pipelines Operator.  Check [here](https://github.com/opendatahub-io/data-science-pipelines-operator) for more information on using the DSPO.

To use and deploy a DSP stack with these images (assuming the DSPO is deployed), first save the following YAML to a file named `dspa.pr-18.yaml`:
```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: pr-18
spec:
  dspVersion: v2
  apiServer:
    image: "quay.io/opendatahub/ds-pipelines-api-server:pr-18"
    argoDriverImage: "quay.io/opendatahub/ds-pipelines-driver:pr-18"
    argoLauncherImage: "quay.io/opendatahub/ds-pipelines-launcher:pr-18"
  persistenceAgent:
    image: "quay.io/opendatahub/ds-pipelines-persistenceagent:pr-18"
  scheduledWorkflow:
    image: "quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-18"
  mlmd:  
    deploy: true  # Optional component
    grpc:
      image: "quay.io/opendatahub/ds-pipelines-metadata-grpc:pr-18"
    envoy:
      image: "quay.io/opendatahub/ds-pipelines-metadata-envoy:pr-18"
  mlpipelineUI:
    deploy: true  # Optional component 
    image: "quay.io/opendatahub/ds-pipelines-frontend:pr-18"
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
```

Then run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines.git
cd data-science-pipelines/
git fetch origin pull/18/head
git checkout -b pullrequest 17c7f1e13fe9020e596e656c124aa5d30b77b296
oc apply -f dspa.pr-18.yaml
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

File: samples/v2/pipeline_container_no_input.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# PIPELINE DEFINITION
+# Name: v2-container-component-no-input
```
Comment (DharmitD) [inline samples/v2/pipeline_container_no_input.yaml:2]: Could we add a README briefly describing each of these pipeline examples? Probably out of scope for this PR/task and should be a follow up task

File: samples/v2/pipeline_with_volume.py [hunk 2]
```diff
@@ -62,7 +62,6 @@ def pipeline_with_volume():
 
 if __name__ == '__main__':
     # execute only if run as a script
-    from kfp import compiler
     compiler.Compiler().compile(
         pipeline_func=pipeline_with_volume,
-        package_path='pipeline_with_volume.json')
\ No newline at end of file
+        package_path='pipeline_with_volume.yaml')
```
Comment (DharmitD) [inline samples/v2/pipeline_with_volume.py:67]: Make sure to add a newline EoF :)

File: samples/v2/component_with_optional_inputs.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# PIPELINE DEFINITION
+# Name: v2-component-optional-input
+components:
```
Comment (DharmitD) [inline samples/v2/component_with_optional_inputs.yaml:3]: We should also add the python code from which each of these yaml specs have been generated.

File: samples/v2/pipeline_with_volume.py (hunk 2 above)
Comment (amadhusu) [reply samples/v2/pipeline_with_volume.py:67]: Man!!! I will add it on the next commit when I add several more IR files here.

File: samples/v2/pipeline_container_no_input.yaml (hunk 1 above)
Comment (amadhusu) [reply samples/v2/pipeline_container_no_input.yaml:2]: Actually this PR itself might get closed if its deemed unnecessary. I like that idea though of having a README briefly describing each of the pipeline examples.

File: samples/v2/component_with_optional_inputs.yaml (hunk 3 above)
Comment (amadhusu) [reply samples/v2/component_with_optional_inputs.yaml:3]: Actually the file name would explain this -> `component_with_optional_inputs.py` is the python code for `component_with_optional_inputs.yaml`

Comment (amadhusu) [reply samples/v2/component_with_optional_inputs.yaml:3]: But I don't see the harm in adding a comment in each of these `.yaml` files again

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-18`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-18`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-18`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-18`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-18`
  **MLMD Server**: `quay.io/opendatahub/ds-pipelines-metadata-grpc:pr-18`
  **MLMD Envoy Proxy**: `quay.io/opendatahub/ds-pipelines-metadata-envoy:pr-18`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-18`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-18`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-18`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-18`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-18`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-18`
  **MLMD Server**: `quay.io/opendatahub/ds-pipelines-metadata-grpc:pr-18`
  **MLMD Envoy Proxy**: `quay.io/opendatahub/ds-pipelines-metadata-envoy:pr-18`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-18`

**Comment by HumairAK** [comment]
Let's not add them here, this would become a carried patch, and this is not the location where we want to add them (though we can propose this upstream first to also include the yaml files)

**Pull request closed without merging**